src/Workspaces/Core/Portable/Workspace/Solution/FilePathToDocumentIdsMap.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// Helper type that keeps track of all the file paths for all the documents in a solution snapshot and all the document
+/// ids each maps to.
+/// </summary>
+internal readonly struct FilePathToDocumentIdsMap
```

Comment:
I wrapped this with a helper type to ensure we understood exactly how it was being used and to be able to audit all consumptino points.